bin/swapSsmId.py

```diff
@@ -5,20 +5,31 @@
 
 if __name__ == '__main__':
 
-    parser = argparse.ArgumentParser(description="Swap ssmId's that are strings into integers.")
+    parser = argparse.ArgumentParser(description="Swap ssmId's that are strings into integers.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
     parser.add_argument("inputFile", type=str, help="Input file containing the DiaSources.")
     parser.add_argument("-t", "--trackId", type=str, default='ssmId_to_int.dat',
                         help="File tracking the map from the ssmId strings to integers.")
     parser.add_argument("-o", "--outFile", type=str, default=None,
                         help="Output file name for diasources with integer ssmIds. "\
                         "Default value is the input file, with addition of '_newId'.")
+    parser.add_argument("-k", "--ssmidKey", type=str, default=None,
+                        help="Name of the input file column containing SSMIDs."\
+                        "If not specified, will use a list of default key names and look"\
+                        "for the correct key.")
     args = parser.parse_args()
 
     diasources = pd.read_table(args.inputFile, delim_whitespace=True)
-    ssmKeys = ['objId', 'ssmId', 'ObjID', 'objID', 'ssmID']
-    for k in ssmKeys:
-        if k in diasources:
-            ssmKey = k
+
+    # If ssmidKey is given on command-line then use that.
+    # If it's not given, use list of defaults and find key.
+    if args.ssmidKey is not None:
+        ssmKey = args.ssmidKey
+    else:
+        ssmKeys = ['objId', 'ssmId', 'ObjID', 'objID', 'ssmID']
+        for k in args.ssmidKeys:
+            if k in diasources:
+                ssmKey = k
```

Comment:
I might be a little confused too. Was the purpose so that the user could provide the ssmIdKey, in case it wasn't included in the defaults?   

And I feel responsible for the particular problems Tim mentioned, as they're problems with my original code ... I should probably have just done the intersection in the first place, and the ssmKey could have been left undefined there too. 

So maybe ... 

if args.ssmidKey is not None:
    ssmKeys = set(args.ssmidKey)
else:
    ssmKeys = set(['objId', 'ssmId', 'ObjID', 'objID', 'ssmID'])
diakeys = set(diasources.columns.values)
if len(diakeys.intersection(ssmKeys)) != 1:
      raise ValueError(something informative about the diakeys and ssmkeys)
ssmKey = diakeys.intersection(ssmKeys).pop()

Or something cleaner than all the set clutter that @timj might suggest.